app/src/main/java/org/mozilla/focus/ScreenController.kt

```diff
@@ -96,9 +96,10 @@ object ScreenController {
         }
     }
 
-    fun showPocketScreen(fragmentManager: FragmentManager) {
+    fun showPocketScreen(fragmentManager: FragmentManager, browserFragment: BrowserFragment) {
         fragmentManager.beginTransaction()
-                .replace(R.id.container, PocketVideoFragment.create(Pocket.getRecommendedVideos()))
+                .hide(browserFragment)
```

Comment:
Interesting, looking into `hide` it's definitely better than `replace` here because we always want to return to the BrowserFragment, so there's no need to tear down BF. I don't *think* there will be bad side effects, since this transaction is in the backstack and can reversed. We also shouldn't have to worry about the the problem where `commit` causes crashes because the Activity is torn down because we basically only have one activity so it'll be alive while we're doing all of this.

Reply:
This could make #1109 more complicated, because if, maybe, we want to show the BrowserFragment on top of the Pocket feed we'd have to make sure to also call `show` on BF because it'll be hidden from this call.

One thing I'm concerned about is (and I don't know what happens when we find a fragment by id): when you click on a video from the Pocket feed, does it show the BF correctly, or is it still hidden? I don't see us ever killing the PocketFragment (remove, replace) so I think....unless we back out of it by tapping "back" it just hangs around forever?

Can you test to make sure it works? but yeah we should fix this blasted architecture now that we have all these screens.

Reply:
@Baron-Severin two things I can think of, if returning to BF doesn't work.

a) updating showBrowserForUrl in `ScreenController` to show the fragment when it's loaded
a) setting the session info in the BF when it's created (but keeping `replace` instead of `hide`/`add`

Reply:
Setting session info kinda works, but we still lose a lot of state in the WebView.  We have to go through loading again, and if the user was midway through filling out a form, or had expanded an image, they'll lose all of that.  

I'm experimenting with changes I can make to `showBrowserForUrl`, but making the entire jigsaw puzzle fit is proving challenging.